src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,138 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private static final double EXPAND_MULTIPLIER = 1.5;
+    private T[] collection;
 
-    @Override
-    public void add(T value) {
+    private int size = 0;
 
+    public ArrayList() {
+        this.collection = (T[]) new Object[CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value) {
+        if (size == collection.length) {
+            expand();
+        }
+        collection[size] = value;
+        size++;
+    }
 
+    @Override
+    public void add(T value, int index) throws ArrayIndexOutOfBoundsException {
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("Index out of bound");
+        }
+        if (size == collection.length) {
+            expand();
+        }
+        T[] listAdd = (T[]) new Object[(size + 1)];
+        for (int i = 0; i < listAdd.length; i++) {
+            if (i < index) {
+                listAdd[i] = collection[i];
+            }
+            if (i == index) {
+                listAdd[i] = value;
+            }
+            if (i > index) {
+                listAdd[i] = collection[i - 1];
+            }
+        }
+        collection = listAdd;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        T[] newList = (T[]) new Object[(list.size() + size)];
+        for (int i = 0; i < size; i++) {
+            newList[i] = collection[i];
+        }
+        for (int i = 0; i < list.size(); i++) {
+            newList[i + size] = list.get(i);
+        }
+        collection = newList;
+        size += list.size();
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public T get(int index) throws ArrayIndexOutOfBoundsException {
```

Comment:
```suggestion
    public T get(int index) {
```
this is Runtime Exception